src/cli/cli.cc

```diff
@@ -1837,68 +1918,139 @@ int main (const int argc, const char* argv[]) {
     std::cout << getPaths(platform.os).pathResourcesRelativeToUserBuild.string() << std::endl;
     exit(0);
   });
 
-  auto const runOptions = Vector<String> {
-    "--platform",
-    "--host",
-    "--port",
-    "--prod",
-    "--test",
-    "--headless"
+  // flag indicating whether option is optional
+  std::map<std::vector<String>, bool> runOptions = {
+    { { "--platform" }, true },
+    { { "--host" }, true },
+    { { "--port" }, true },
+    { { "--prod" }, true },
+    { { "--test" }, true },
+    { { "--headless" }, true }
   };
 
-  auto buildOptions = Vector<String> {
-    "--quiet",
-    "-o",
-    "--only-build",
-    "-r",
-    "--run",
-    "-p",
-    "-c",
-    "-s",
-    "-e",
-    "-n"
+  std::map<std::vector<String>, bool> buildOptions = {
+    { { "--quiet" }, true },
+    { { "--only-build", "-o" }, true },
+    { { "--run", "-r" }, true },
+    { { "-p" }, true },
+    { { "-c" }, true },
+    { { "-s" }, true },
+    { { "-e" }, true },
+    { { "-n" }, true }
   };
 
-    // Insert the elements of runOptions into buildOptions
-  buildOptions.insert(buildOptions.end(), runOptions.begin(), runOptions.end());
-  createSubcommand("build", buildOptions, true, [&](const std::span<const char *>& options) -> void {
-    bool flagRunUserBuildOnly = false;
-    bool flagAppStore = false;
-    bool flagCodeSign = false;
-    bool flagHeadless = false;
-    bool flagShouldRun = false;
-    bool flagEntitlements = false;
-    bool flagShouldNotarize = false;
-    bool flagShouldPackage = false;
+  // Insert the elements of runOptions into buildOptions
+  buildOptions.insert(runOptions.begin(), runOptions.end());
+  createSubcommand("build", buildOptions, true, [&](Map argumentsWithValue, std::unordered_set<String> argumentsWithoutValue) -> void {
+    String argvForward = "";
+    String targetPlatform = argumentsWithValue["--platform"];
+    bool flagRunUserBuildOnly = argumentsWithoutValue.find("--only-build") != argumentsWithoutValue.end() || equal(rc["build_only"], "true");
+    bool flagAppStore = argumentsWithoutValue.find("-s") != argumentsWithoutValue.end() || equal(rc["build_app_store"], "true");
+    bool flagCodeSign = argumentsWithoutValue.find("-c") != argumentsWithoutValue.end() || equal(rc["build_codesign"], "true");
+    bool flagHeadless = argumentsWithoutValue.find("--headless") != argumentsWithoutValue.end() || equal(rc["build_headless"], "true");
+    bool flagShouldRun = argumentsWithoutValue.find("--run") != argumentsWithoutValue.end() || equal(rc["build_run"], "true");
+    bool flagEntitlements = argumentsWithoutValue.find("-e") != argumentsWithoutValue.end() || equal(rc["build_entitlements"], "true");
+    bool flagShouldNotarize = argumentsWithoutValue.find("-n") != argumentsWithoutValue.end() || equal(rc["build_notarize"], "true");
+    bool flagShouldPackage = argumentsWithoutValue.find("-p") != argumentsWithoutValue.end() || equal(rc["build_package"], "true");
```

Comment:
Again, this is quite verbose and there's a lot of repeated code that could be refactored.

Reply:
100%

I want to split this work into a few PRs. It's already hard to review this one (though hopefully reviewing by commits can help) and there will be many conflicts if someone will touch `cli.cc` too. There are still many things that can be simplified. It's just a first milestone.